### PR TITLE
Add TextRender usage documentation

### PR DESCRIPTION
**Pull request by stdweird**
This is the `CAF::TextRender` part of the tutorial given during the 19th workshop.

### CODE REVIEW

File: TextRender.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
+# TextRender
+
+Generating structured text is best done with [`CAF::TextRender`][caf_textrender_docs].
+This document guides through the usage and testing of `CAF::TextRender`.
+
+Using `ncm-metaconfig`, which is the metacomponent build around
```
Comment (jrha) [inline TextRender.md]: buil**t**

File: TextRender.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+[caf_textrender_docs]: http://docs-test-caf.readthedocs.org/en/latest/CAF/CAF::TextRender
+[metaconfig]: https://github.com/quattor/configuration-modules-core/Metaconfig.md
+
+# CAF::TextRender
+
+Basic usage has 2 main modes:
+ * generate text : the `CAF::TextRender` instance has auto-stringification
+
+    <!-- language: lang-perl -->
```
Comment (jrha) [inline TextRender.md]: Does this work? Shouldn't it use \`\`\`perl for consistency?

File: TextRender.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        use CAF::TextRender;
+        my $module = 'mymodule';
+        my $trd = CAF::TextRender->new($module, $contents, log => $self);
+        print "$trd"; # stringification
+
+ * write text to file : get a `CAF::FileWriter` instance with text from `CAF::TextRender` instance
+
+    <!-- language: lang-perl -->
```
Comment (jrha) [inline TextRender.md]: Same question as above.

File: TextRender.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        use CAF::TextRender;
+        $module = "mymodule";
+        $trd = CAF::TextRender->new($module, $contents, log => $self);
+        my $fh = $trd->filewriter('/some/path');
+        die "Problem rendering the text" if (!defined($fh));
+        $fh->close();
+
+Besides the logger, the 2 main parameters are the `module` and the `contents`.
+The contents is a hash-reference with the data that is used to generate
```
Comment (jrha) [inline TextRender.md]: \` around contents

File: TextRender.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+```
+
+will generate
+
+```
+$ perl -e 'use Template; my $tttext="Hello [% world %]\n"; Template->new()->process(\$tttext, { world => "Quattor" });'
+Hello Quattor
+```
+
+TODO minimal version
```
Comment (jrha) [inline TextRender.md]: Any idea what this is yet?

File: TextRender.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+
+[TT_home]: http://www.template-toolkit.org/index.html
+
+TODO add url with examples
```
Comment (jrha) [inline TextRender.md]: How about https://github.com/quattor/configuration-modules-core/tree/master/ncm-metaconfig/src/main/metaconfig as a start?

File: TextRender.md [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+    );
+$rt->test();
+```
+
+With the default flags, each line is compiled as a multiline regular expression and matched against the text.
+The matches are also checked if they are ordered. In the example above `line 3` is expected to match in the text
+following `line 1`. But it does not need to be the next line (e.g. there could be a `line 2` in between).
+Each match is a test and each verification of the ordering also.
+
+TODO add correct/final url
```
Comment (jrha) [inline TextRender.md]: ?

**Comment by jouvin** [comment]
It would be good to have it finalized at the time of 15.4, even if not strictly part of the release...

File: TextRender.md (hunk 2 above)
Comment (stdweird) [inline TextRender.md]: That's for gihhub markdown, but generating documentation for readthedocs does understand github tags.

Comment (stdweird) [inline TextRender.md]: i'll add the gh tags, we can figure out the readthedocs issue later

Comment (stdweird) [inline TextRender.md]: i'm keeping the xml tags for now
see https://github.com/quattor/release/issues/86
this is each to search and replace later

File: TextRender.md (hunk 7 above)
Comment (stdweird) [inline TextRender.md]: the idea was to move all the readthe docs under one url, not one per repo. but i'll leave it as is

**Comment by stdweird** [comment]
@jrha i'm adding a CCM::TextRender section too

**Comment by stdweird** [comment]
@jrha i added the `CCM::TextRender` section, including an example that will only work if some of my branches get merged in 15.6 (in particular the `path` magic) (but are the best example of what `CCM::TextRender` can do)

File: TextRender.md (hunk 2 above)
Comment (jrha) [inline TextRender.md]: does understand github tags?
or 
doesn't understand github tags.

**Comment by stdweird** [comment]
@jrha afaik, readthedocs does not understand gh tags (but i removed the tags to be consistent with the rest of the doc. i opened https://github.com/quattor/release/issues/86 to find a solution

**Comment by jrha** [comment]
:crying_cat_face:

**Pull request merged**